src/main/kotlin/no/nav/tilleggsstonader/sak/vilkår/vilkårperiode/domain/faktavurderinger/FaktaOgVurdering.kt

```diff
@@ -51,49 +51,9 @@ data object IngenFakta : Fakta
  * Vurderinger, som kan inneholde ulike vurderinger, eks lønnet
  */
 sealed interface Vurderinger {
-    fun resultatVurderinger(): ResultatVilkårperiode {
-        val resultater = finnResultatetFraVurderinger()
-        return utledResultat(resultater)
-    }
-
-    private fun finnResultatetFraVurderinger(): MutableList<ResultatDelvilkårperiode> {
-        val resultater = mutableListOf<ResultatDelvilkårperiode>()
-        if (this is LønnetVurdering) {
-            resultater.add(lønnet.resultat)
-        }
-        if (this is MedlemskapVurdering) {
-            resultater.add(medlemskap.resultat)
-        }
-        if (this is DekketAvAnnetRegelverkVurdering) {
-            resultater.add(dekketAvAnnetRegelverk.resultat)
-        }
-        if (this is HarUtgifterVurdering) {
-            resultater.add(harUtgifter.resultat)
-        }
-        if (this is HarRettTilUtstyrsstipendVurdering) {
-            resultater.add(harRettTilUtstyrsstipend.resultat)
-        }
-        return resultater
-    }
-
-    private fun utledResultat(resultater: List<ResultatDelvilkårperiode>) =
-        when {
-            resultater.contains(ResultatDelvilkårperiode.IKKE_VURDERT) -> {
-                ResultatVilkårperiode.IKKE_VURDERT
-            }
-
-            resultater.contains(ResultatDelvilkårperiode.IKKE_OPPFYLT) -> {
-                ResultatVilkårperiode.IKKE_OPPFYLT
-            }
-
-            resultater.all { it == ResultatDelvilkårperiode.OPPFYLT } -> {
-                ResultatVilkårperiode.OPPFYLT
-            }
-
-            else -> {
-                error("Ugyldig resultat ($resultater)")
-            }
-        }
+    fun utledResultat(): ResultatVilkårperiode
```

Comment:
Et alternativ er å ha en metode som returnerer listen for alle delvilkår.
```kotlin
fun alleDelvilkår: List<ResultatDelvilkår>`

//og 
fun  utledResultat(): ResultatVilkårperiode = utledResultatVilkårperiode(alleDelvilkår())
```

Reply:
Det er absolutt mulig, men er litt usikker på hvilke fordel det gir. Hadde du noe i tankene?

Reply:
Hmm! Jeg får går i tenkeboksen